src/generate/generatePdf.ts

```diff
@@ -17,18 +18,170 @@ export interface GeneratePDFResult {
     error?: Error;
 }
 
+interface TOCEntry {
```

Comment:
let's move the logic for creating the table of contents and bookmarks to a separate file, this one is getting too big